"TypeError: int() argument must be a string" with ASGI + Django:

[NOTE]: #
Since instrumenting my Django application, I started seeing these errors:
```
[ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 411, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 69, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/newrelic/api/asgi_application.py", line 357, in nr_async_asgi
    return await coro
           ^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/newrelic/common/async_proxy.py", line 124, in throw
    return self.__wrapped__.throw(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/newrelic/api/asgi_application.py", line 69, in __call__
    return await self.app(scope, receive, self.send_inject_browser_agent)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/channels/routing.py", line 62, in __call__
    return await application(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/django/asgi.py", line 101, in sentry_patched_asgi_handler
    return await middleware(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/asgi.py", line 144, in _run_asgi3
    return await self._run_app(scope, receive, send, asgi_version=3)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/asgi.py", line 235, in _run_app
    raise exc from None
  File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/asgi.py", line 230, in _run_app
    return await self.app(
           ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/newrelic/api/asgi_application.py", line 307, in nr_async_asgi
    return await wrapped(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/asgi.py", line 170, in __call__
    await self.handle(scope, receive, send)
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/asgi.py", line 216, in handle
    task.result()
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/asgi.py", line 195, in process_request
    await self.send_response(response, send)
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/asgi.py", line 357, in send_response
    await send(
  File "/usr/local/lib/python3.11/site-packages/sentry_sdk/integrations/asgi.py", line 223, in _sentry_wrapped_send
    return await send(event)
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/newrelic/api/asgi_application.py", line 177, in send_inject_browser_agent
    content_length = int(header_value)
                     ^^^^^^^^^^^^^^^^^
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
```

**Description**
[NOTE]: # ( Describe the problem you're encountering. )
This error occurs only on some requests.
[TIP]:  # ( Do NOT share sensitive information, whether personal, proprietary, or otherwise! )

**Expected Behavior**
[NOTE]: # ( Tell us what you expected to happen. )
No error happening, which was the case before I installed the SDK.

**[Troubleshooting](https://discuss.newrelic.com/t/troubleshooting-frameworks/108787) or [NR Diag](https://docs.newrelic.com/docs/using-new-relic/cross-product-functions/troubleshooting/new-relic-diagnostics) results**
[NOTE]: # ( Provide any other relevant log data. )
[TIP]:  # ( Scrub logs and diagnostic information for sensitive information )

**Steps to Reproduce**
[NOTE]: # ( Please be as specific as possible. )
1. Install `sentry_sdk`
2. Make an HTTP request
[TIP]:  # ( Link a sample application that demonstrates the issue. )

**Your Environment**
[TIP]:  # ( Include as many relevant details about your environment as possible including the running version of New Relic software and any relevant configurations. )
Run command:
```
[”/usr/local/bin/gunicorn”, “–bind”, “0.0.0.0:8000”, “–workers”, “1”, “–threads”, “8”, “–timeout”, “0”, “example.asgi:application”]
```

**Runtime**
Name: CPython
Version: 3.11.9(3.11.9 (main, May 14 2024, 06:28:33) [GCC 12.2.0])


### Packages

| Package                     | Version   |
|-----------------------------|-----------|
| aiofiles                    | 23.2.1    |
| aiohttp                     | 3.9.4     |
| asgiref                     | 3.7.2     |
| channels                    | 4.1.0     |
| channels-redis              | 4.2.0     |
| django                      | 5.0.6     |
| django-allauth              | 0.62.1    |
| django-allauth-2fa          | 0.11.1    |
| django-cors-headers         | 4.3.1     |
| django-environ              | 0.11.2    |
| django-hijack               | 3.4.5     |
| django-htmx                 | 1.17.3    |
| django-otp                  | 1.3.0     |
| django-redis                | 5.4.0     |
| django-storages             | 1.14.3    |
| django-timezone-field       | 6.1.0     |
| django-waffle               | 4.1.0     |
| djangorestframework         | 3.15.1    |
| gunicorn                    | 22.0.0    |
| httptools                   | 0.6.1     |
| httpx                       | 0.27.0    |
| newrelic                    | 9.9.0     |
| psycopg                     | 3.1.18    |
| psycopg-c                   | 3.1.18    |
| sqlparse                    | 0.4.4     |
| uvicorn                     | 0.29.0    |
| uvloop                      | 0.19.0    |
| websockets                  | 12.0      |
| sentry.python.django        | 2.1.1     |


**Additional context**
[TIP]:  # ( Add any other context about the problem here. For example, relevant community posts or support tickets. )

--- a/newrelic/api/asgi_application.py
+++ b/newrelic/api/asgi_application.py
@@ -175,7 +175,7 @@ class ASGIBrowserMiddleware(object):
 
                     try:
                         content_length = int(header_value)
-                    except ValueError:
+                    except (TypeError, ValueError):
                         # Invalid content length results in an abort
                         await self.send_buffered()
                         return

--- a/tests/agent_features/test_asgi_browser.py
+++ b/tests/agent_features/test_asgi_browser.py
@@ -688,7 +688,7 @@ def test_html_insertion_invalid_content_length():
 
 
 @override_application_settings(_test_html_insertion_invalid_content_length_settings)
-def test_html_insertion_invalid_content_length():
+def test_html_insertion_no_content_length():
     response = target_application_no_content_length.get("/")
     assert response.status == 200
 

--- a/tests/agent_features/test_asgi_browser.py
+++ b/tests/agent_features/test_asgi_browser.py
@@ -653,6 +653,19 @@ async def target_asgi_application_invalid_content_length(scope, receive, send):
 
 target_application_invalid_content_length = AsgiTest(target_asgi_application_invalid_content_length)
 
+
+@asgi_application()
+async def target_asgi_application_no_content_length(scope, receive, send):
+    output = b"<html><body><p>RESPONSE</p></body></html>"
+
+    response_headers = [(b"content-type", b"text/html; charset=utf-8")]
+
+    await send({"type": "http.response.start", "status": 200, "headers": response_headers})
+    await send({"type": "http.response.body", "body": output})
+
+
+target_application_no_content_length = AsgiTest(target_asgi_application_no_content_length)
+
 _test_html_insertion_invalid_content_length_settings = {
     "browser_monitoring.enabled": True,
     "browser_monitoring.auto_instrument": True,
@@ -669,6 +682,17 @@ def test_html_insertion_invalid_content_length():
     assert "content-length" in response.headers
 
     assert response.headers["content-length"] == "XXX"
+
+    assert b"NREUM HEADER" not in response.body
+    assert b"NREUM.info" not in response.body
+
+
+@override_application_settings(_test_html_insertion_invalid_content_length_settings)
+def test_html_insertion_invalid_content_length():
+    response = target_application_no_content_length.get("/")
+    assert response.status == 200
+
+    assert "content-type" in response.headers
 
     assert b"NREUM HEADER" not in response.body
     assert b"NREUM.info" not in response.body

